### ISSUE TITLE
Popup are overlapping

### ISSUE THREAD

**Issue by mathieu-fournier** [iTwin/appui#1135]
### Describe the bug

To the left, it's the expected behaviour, which usually happens. 
In the center, we can see that FrameworkToolAdmin element tooltip overlaps the tool assistance popup. It's hidden underneat. 
To the right, we can see that popup manager can have overlapping popups. This happens when the focus is lost in the view, then taken again. Doing this stops tool assistance, then resumes it, appearing on to of the already-existing popup. 

### Screenshots

![Image](https://github.com/user-attachments/assets/7c5ceefb-55a0-44ea-ae24-71ebe2f6493e)

**Comment by simnorm** [comment]
@GerardasB  also required for OpenSite+.  Who will look at supporting this?

**Comment by GerardasB** [comment]
Currently each individual cursor popup and element tooltip is rendered as a separate element on the screen. I.e. all cursor popups are positioned individually, rather than rendered in a container which is later positioned.

Some known positioning constraints:
All cursor popups share same cursor "position".
Element tooltip can be displayed at any given point.

I'm proposing to address this issue in 2 separate steps:
1. Render no cursor popups when element tooltip is displayed
2. Update positioning logic of cursor popups to avoid overlaps (initially I thought we could render these in a shared container, but it might be too tricky given the extensive customization options - offset, relative position, priority).

**Comment by mathieu-fournier** [comment]
@GerardasB 
I agree with you, this doesnt seem like something appui can fix alone. 
Since components at the cursor can come from any apps and packages. Even if we make very complex code to avoid known overlaps, any app or package can introduce a new cursor component and we'll have the overlap again.
We can do one of the options has a temporary fix. 

@mayank99 
This might be something ITwinUI could provide us has a generic component ?
Then all app and package could use the same component that would take care of preventing overlaps and to place things properly.
My naive approach would be to use [css grid-positionning](https://www.creativebloq.com/advice/a-comprehensive-guide-to-using-css-grid) so it would adjust itself automatically and support many displays. 
Pseudo-code example: 
`const gridPosition : CssGridProps = ...someprops;`
`<CursorComponent position={gridPosition}><div>cursor potato<div/><CursorComponent/>`
